lang/ruby-devel/Portfile

```diff
@@ -125,7 +128,14 @@ platform darwin {
         depends_build-append    port:gmake
         build.cmd               ${prefix}/bin/gmake
         configure.args-append   --disable-dtrace
-        configure.cflags-append -std=c99
+        # Do not pass -std=c99, it breaks lgamma_r:
+        # math.c: error: implicit declaration of function 'lgamma_r';
```

Comment:
When you make fixes like this, it might be reasonable to apply them only to the single file or files that need it. That said, I presume that the error is noting an actual problem, which probably could be patched.

Reply:
Well, the default assumption should be that build system sets correct standard where needed. We should not set a specific standard unnecessarily. Here it just breaks things.
It is not the first case of such, `-std=c99` has broken some builds with clang elsewhere.

Reply:
The point is, usually fixing the problem is a one line patch or the like to the source file, which can then be contributed upstream. Changing the compile flags is a hack.

Reply:
It is our flag which is wrong, not the source. Source code compiles fine, once the flag is dropped.

Reply:
Implicit declarations aren't a good thing, they result in code defects. There's a good reason not to want code with implicit declarations in one's codebase. Similarly, code that isn't up to C99 is a long term problem, especially as pre-C99 constructs are being deprecated. Generally, if only one or two files in a codebase are messed up, there's a good reason to fix them.

Reply:
There are two separate issues here, and removal of the flag is related only to one of those.

For `fgetattrlist` a better solution would be to provide a declaration when it is missing, or rather do it MacPorts-wide via `legacy-support`. But it is not something I am ready to do myself, because I do not know how to test it. Of course, copy-pasting from 10.6.8 header isn’t hard.
This is unrelated to C99.

For `lgamma_r` there is no problem with the source code, and Apple headers have it declared. But our `-std=c99` flag obscures it. Why? I have no idea. But I cannot file a bug to upstream saying that the code does not behave as expected when MacPorts adds a custom flag. There is an obvious solution: do not add a flag which messes things up, setting an archaic C standard artificially.

Reply:
I suspect the issue is that there is (in fact) a subtle problem with the source code. Look at the header declaration of lgamma_r on both the platforms where it works and where it doesn't. Unfortunately I don't have a <10.7 system to look at.

Reply:
It “worked” because the compiler did not raise it to an error:
```
math.c:1035:17: warning: implicit declaration of function 'lgamma_r' is invalid in C99 [-Wimplicit-function-declaration]
    v = DBL2NUM(lgamma_r(d, &sign));
                ^
1 warning generated.
```
https://build.macports.org/builders/ports-10.6_x86_64-builder/builds/208403/steps/install-port/logs/stdio

It was the same for me locally until a move to `gcc14`, which makes it an error.

Since `-std=c99` was only used for < 10.7, later systems are irrelevant.

Reply:
Right, so, why not patch it so that you explicitly declare the function?

Reply:
If you refer to `lgamma_r`, it should come from `libm`, where it is already present. The bug is ours, we break it by downgrading C standard.

The second function I have requested a help with, see below. If that is done, then we will not need any configure-time tweaks.